app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb

```diff
@@ -101,6 +101,42 @@ def self.params_for_create(ems)
             }
           end,
         },
+        {
+          :component    => 'select',
+          :name         => 'affinityPVMInstance',
+          :id           => 'affinityPVMInstance',
+          :label        => _('Cloud Volume Type'),  
+          :validate     => [{:type => 'required'}],
+          :condition    => {
+            :when    => 'affinity_policy',
+            :pattern => '^Off$',
+          },
```

Comment:
There needs to be an additional field to select affinity type: Volume or PVM Instance. Then the volume and VM selection fields can be conditional on the affinity type selected.